recipes/archicad-apidevkit/all/conanfile.py

```diff
@@ -0,0 +1,52 @@
+from conans import tools
+from conan import ConanFile
+from conan.tools.files import copy, get
+from conan.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.46.0"
+
+
+class ArchicadApidevkitConan (ConanFile):
+    name = "archicad-apidevkit"
+    description = "The General API Development Kit enables software developers to extend the functionality of Archicad"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://archicadapi.graphisoft.com/"
+    license = "LicenseRef-LICENSE"
+    settings = "os", "arch", "build_type", "compiler"
+    no_copy_source = True
+    topics = "api", "archicad", "development"
+    short_paths = True
+
+    @property
+    def _acdevkit_arch(self):
+        return str(self.settings.arch)
+
+    def package_id(self):
+        del self.info.settings.compiler
+        del self.info.settings.build_type
+
+    def package_info(self):
+        self.cpp_info.bindirs = []
+        self.cpp_info.frameworkdirs = []
+        self.cpp_info.libdirs = []
+        self.cpp_info.includedirs = []
+
+    def validate(self):
+        if not self.info.settings.os in ("Macos", "Windows"):
+            raise ConanInvalidConfiguration(
+                f"{self.ref} is not supported by the OS {self.info.settings.os}")
+        if not str(self.settings.arch) in ("x86_64"):
+            raise ConanInvalidConfiguration(
+                f"{self.ref} is not supported yet.")
```

Comment:
Since the VS toolset is also a requirement, you should also be checking the toolset in the config when the MSBuild build system is in use: https://docs.conan.io/en/latest/integrations/build_system/msbuild.html#toolsets

I can't find an example for this in CCI, so I don't know how to check for the build system here.

Reply:
It's a new setting that by default is none in 1.x but should be present for the 2.0 pipline when that rolles out

You can check the compiler's version in the mean time we have `check_min_msvc` for this